tests/integration/bond_test.py

```diff
@@ -265,3 +265,81 @@ def test_set_bond_mac_address(eth1_up):
         eth1_cur_state = current_state[INTERFACES][1]
         assert bond99_cur_state[Interface.MAC] == MAC1.upper()
         assert bond99_cur_state[Interface.MAC] == eth1_cur_state[Interface.MAC]
+
+
+@pytest.fixture
+def bond99():
+    with bond_interface(name='bond99', slaves=[]) as bond_state:
+        yield bond_state
+
+
+def test_bond_and_test_inferfaces_have_different_macs(bond99):
+    slaves_state = statelib.show_only(('bond99', 'eth1', 'eth2'))
+    bond99_slaves_state = slaves_state[Interface.KEY][0]
+    eth1_slaves_state = slaves_state[Interface.KEY][1]
+    eth2_slaves_state = slaves_state[Interface.KEY][2]
+
+    bond99_slaves_mac = bond99_slaves_state[Interface.MAC]
+    eth1_slaves_mac = eth1_slaves_state[Interface.MAC]
+    eth2_slaves_mac = eth2_slaves_state[Interface.MAC]
+
+    assert bond99_slaves_mac != eth1_slaves_mac
+    assert eth1_slaves_mac != eth2_slaves_mac
+    assert eth2_slaves_mac != bond99_slaves_mac
+
+
+def test_bond_and_all_removed_slaves_have_same_mac(bond99):
```

Comment:
Don't have this test anymore, it was written so comply with your steps(in comments) on Jira issue 198 and to add eth2 since  test_remove_all_slaves_from_bond didn't consider eth2